fargate-execution-role.tf

```diff
@@ -9,7 +9,9 @@ data "aws_iam_policy_document" "ecs_execution_principal" {
   }
 }
 locals {
-  extracted_container_secrets = flatten([for c in local.container_definitions : try(c.secrets, [])])
+  _extracted_container_secrets = flatten([for c in local.container_definitions : try(c.secrets, [])])
+  //strips the secret-name:json-key:version-stage:version-id suffix from the valueFrom field when it's a secret https://docs.aws.amazon.com/AmazonECS/latest/developerguide/secrets-envvar-secrets-manager.html
```

Comment:
@mikewolfd can you explain this a little bit more in detail as to what is happening here? Can you show an example of what this is doing?

Reply:
Thanks for making me look at this again Lol, I made it a little simpler/better and fixed an edgecase.

Pretty much the issue is that for secrets there's a special `json-key:version-stage:version-id` syntax that you can use in task container definitions for extracting the values out of the secrets objects directly, instead of having to pass them into the container. Unforunately AWS's IAM engine doesn't parse it correctly, thus making it so ECS is unable to access the secret. This presents with minimal logging, except for a random orange alert in the web console.

This function just checks to see if a secret is using this syntax by seeing if there's extra colons, i.e. the syntax requires the extra colons. Like `<arn>:json-key:stage:` is an acceptable syntax, it requires all 3 colons. The function then drops them and rejoins the string.